client/extensions/woocommerce/components/reading-widget/index.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import emailValidator from 'email-validator';
+import { localize } from 'i18n-calypso';
+import React, { Component, PropTypes } from 'react';
+import ExternalLink from 'components/external-link';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import FormTextInput from 'components/forms/form-text-input';
+import { getSitePostsForQueryIgnoringPage } from 'state/posts/selectors';
+import humanDate from 'lib/human-date';
+import MultiCheckbox from 'components/forms/multi-checkbox';
+import QueryPosts from 'components/data/query-posts';
+
+class ReadingWidget extends Component {
+
+	state = {
+		emailAddress: '',
+		roles: [ 'storeOwner' ],
+		subscriptionFormExpanded: false,
+	};
+
+	static propTypes = {
+		query: PropTypes.object,
+		siteId: PropTypes.number,
+		text: PropTypes.string,
+		title: PropTypes.string,
+	};
+
+	static defaultProps = {
+		query: { category: 'start-your-store', number: 3 },
+		siteId: 113771570,
+	}
+
+	getRolesAndLabels = () => {
+		const { translate } = this.props;
+
+		return [
+			{ value: 'storeOwner', label: translate( 'Store owner' ) },
+			{ value: 'developer', label: translate( 'Developer' ) },
+			{ value: 'extensionDeveloper', label: translate( 'Extension developer' ) },
+			{ value: 'other', label: translate( 'Other' ) },
+		];
+	};
+
+	handleSubscriptionEmailFocus = () => {
+		this.setState( { subscriptionFormExpanded: true } );
+	}
+
+	renderHeading = () => {
+		const { text, title } = this.props;
+		return (
+			<div className="reading-widget__heading">
+				<h2>{ title }</h2>
+				<p>{ text }</p>
+			</div>
+		);
+	}
+
+	renderArticleList = () => {
+		const { posts, query, siteId } = this.props;
+
+		return (
+			<div>
+				<QueryPosts
+					query={ query }
+					siteId={ siteId }
+				/>
+				<ul className="reading-widget__article-list">
+					{
+						( posts && posts.map( post => {
+							const { date, ID, title, URL } = post;
+							return (
+								<li key={ ID }>
+									<ExternalLink icon={ true } href={ URL } rel="noopener noreferrer" target="_blank">
+										{ title }
+									</ExternalLink>
+									<span>{ humanDate( date ) }</span>
+								</li>
+							);
+						} ) )
+					}
+				</ul>
+			</div>
+		);
+	}
+
+	renderSubscriptionFormExplanation = ( expanded ) => {
+		const { translate } = this.props;
+
+		return (
+			<div className="reading-widget__subscription-form-explanation">
+				<p>
+					{ translate( 'Subscribe to our newsletter and get the latest' +
+						' news delivered to your inbox' ) }
+				</p>
+				{
+					expanded && ( <p>
```

Comment:
I think I'd like to decline this suggestion - I think `renderSubscriptionFormExplanation` reads well as is